### PR TITLE
Define scale properties in main config

### PR DESCRIPTION
**Pull request by ericpromislow**
## WHAT is this change about?

https://www.pivotaltracker.com/story/show/174586782

## Does this PR introduce a change to `config/values.yml`?
Yes (`config/values/00-values.yml`)

## Acceptance Steps

This change is mainly to define changes to the interface. It's sufficient to deploy this branch and run smoke tests successfully.

## Tag your pair, your PM, and/or team
Eric Promislow & @davewalter 

## Things to remember
[x] Make sure this PR is based off the `develop` branch
[x] Include any links to related PRs, issues, stories, slack discussions, etc... that will help establish context.
[x] No: Is there anything else of note that the reviewers should know about this change?
[x] None: - Our CI uses concourse task files from HEAD of the `develop` branch. If your PR includes/requires CI changes, please ping the RelInt interrupt and they'll help apply the CI changes for you.
[x] Read: - Checkout the [contributing guidelines](https://github.com/cloudfoundry/cf-for-k8s/blob/develop/community/CONTRIBUTING.md)

### CODE REVIEW

File: config/capi/capi.yml [hunk 1]
```diff
@@ -137,3 +137,22 @@ metadata:
...
+#@overlay/match by=overlay.subset({"kind":"Deployment", "metadata":{"name":"cf-api-server"}})
+---
+#@overlay/match-child-defaults missing_ok=True
+spec:
+  replicas: #@ data.values.capi.scale.cf_api_server.replicas
+  template:
+    spec:
+      containers:
+      - #@overlay/match by=overlay.subset({"name":"cf-api-server"})
+        resources:
```
Comment (loewenstein) [inline config/capi/capi.yml:151]: If we want to allow QoS guaranteed I think we'll have to expose every single container in the deployment.

Comment (davewalter) [reply config/capi/capi.yml:151]: Hi @loewenstein. Thanks for the feedback. We are still working on implementing all of the required overlays to make use of the scaling configuration we added to `config/values/00-values.yml`. We wanted to get this submitted quickly to get more eyes on the proposed user interface, but recognise that the work is a long way from being complete and ready to merge. I have switched the PR to a draft in the meantime to reflect this.

Comment (loewenstein) [reply config/capi/capi.yml:151]: Hi @davewalter,

I realized that after commenting, when I had seen the proposed values. 

Might be worth checking with @cppforlife if we could traverse the data values structure and create overlays for each container without all the repetition.

**Comment by cppforlife** [comment]
per @loewenstein request, here is more generic version: https://get-ytt.io/#gist:https://gist.github.com/cppforlife/32d6ed87a18c66333dab8e710e180dea

**Comment by davewalter** [comment]
Closing as we have decided not to implement this interface at this time.

**Pull request closed without merging**